tests/interaction_test.py

```diff
@@ -372,22 +372,6 @@ def test_basic_question_is_valid__no_validator__always_valid(value):
     assert result.valid
 
 
-@pytest.mark.parametrize(
-    ["is_valid", "validator_response"],
-    [(True, None), (False, "some-error")],
-)
-def test_basic_question_is_valid__legacy_validator__result_of_validator(
```

Comment:
Maybe convert this to be a test case that confirms that returning a string raises `ValueError`.

Reply:
It actually won't return a `ValueError` here, `BasicQuestion.validate()` will return the string if the type signature for a validator isn't respected. This won't blow up until it is called by `_cli.py` at line 182. If you want I can either put defensive code that looks for string returns and raises, or put a test in `_cli.py` that it raises. 

I feel it would be a little strange to do either; like we have type hints to show the correct signature, not sure that we should validate that our code crashes when people use it incorrectly, or add defensive code for that case when it is already clear from the type signature and documentation.

Reply:
It would be like 

```python
        if callable(self._validator):
            result = self._validator(value, answers)
            if isinstance(result, str):
                raise ValueError('you did it wrong')
            return result
            
```

Reply:
Hmm. I meant a test case that columbia no longer works with the old signature. But I guess the error would be a level up in where `validate()` is used.

It is a little weird because it is asserting that something no longer works. So it isn't a requirement to add.